Mathlib/RingTheory/MvPolynomial/MonomialOrder.lean

```diff
@@ -469,4 +469,34 @@ theorem isUnit_leadingCoeff {f : MvPolynomial σ R} :
 
 end Field
 
+section Binomial
+
+variable {R : Type*} [CommRing R]
+
+open Finsupp MvPolynomial
+
+lemma degree_X_sub_C [Nontrivial R]
+    {ι : Type*} (m : MonomialOrder ι) (i : ι) (r : R) :
+    m.degree (X i - C r) = single i 1 := by
+  rw [degree_sub_of_lt, degree_X]
+  simp only [degree_C, map_zero, degree_X]
+  rw [← bot_eq_zero, bot_lt_iff_ne_bot, bot_eq_zero, ← map_zero m.toSyn]
+  simp
+
+lemma leadingCoeff_X_sub_C {ι : Type*} (m : MonomialOrder ι) (i : ι) (r : R) :
+    m.leadingCoeff (X i - C r) = 1 := by
```

Comment:
Do we not have `MonomialOrder.Monic`?

Reply:
No. I can add it, following what is done for `Polynomial.Monic`.

Reply:
Great! (in a separate PR, pretty please 🙏)

Reply:
but probably a new PR would be a better fit.

Reply:
#21546
I added a lot about this, but I'm not sure I wish to do much more.